src/sonic-pac/paccfg/pac_authmgrcfg.cpp

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2019 Broadcom Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <swss/logger.h>
+#include <swss/dbconnector.h>
+#include <swss/producerstatetable.h>
+#include <swss/macaddress.h>
+#include <swss/tokenize.h>
+#include "pac_authmgrcfg.h"
+
+using namespace std;
+using namespace swss;
+
+// PAC SONIC config engine 
+PacCfg::PacCfg(DBConnector *db, DBConnector *cfgDb, DBConnector *stateDb) :
+    m_cfgFdbTable(cfgDb, CFG_FDB_TABLE_NAME),
+    m_stateOperFdbTable(stateDb, STATE_OPER_FDB_TABLE_NAME),
+    m_stateOperPortTable(stateDb, STATE_OPER_PORT_TABLE_NAME)
+{
+    Logger::linkToDbNative("paccfg");
+    SWSS_LOG_NOTICE("PAC: config object");
+
+    /* FDB flush notification producer */
+    m_flushFdb = std::make_shared<swss::NotificationProducer>(db, "FLUSHFDBREQUEST");
+}
+
+PacCfg::~PacCfg()
+{
+
+}
+
+// Set learning mode for a port 
+bool PacCfg::intfLearningSet(string port, string learning)
+{
+   string key(port);
+   vector<FieldValueTuple> fvVector;
+  
+   // Configure port learning mode for FDB manager 
+   fvVector.emplace_back("learn_mode", learning);
+   m_stateOperPortTable.set(key, fvVector);
```

Comment:
This sets the value of the learning mode in the STATE_DB of PAC. Should we update this to the APPL_DB of port table also so that it is pushed down

Reply:
PAC doesn't interact directly with the Orch agent. The Orch agent updates are done using VLAN mgr.  Hence PAC uses STATE_DB .